pkg/prometheus/common.go

```diff
@@ -474,3 +478,39 @@ func BuildStatefulSetService(name string, selector map[string]string, p monitori
 
 	return svc
 }
+
+// This function is responsible for the following:
+// If a ServiceName is specified, verify that the custom governing service exists in the namespace.
+// If it does not exist, fail the reconciliation.
+// Also, ensure that the Prometheus instance is selected by the custom governing service.
+// If it is not selected, fail the reconciliation.
+// If a ServiceName is not specified, create or update the default governing service.
+func CheckCustomGoverningService(ctx context.Context, prometheus monitoringv1.PrometheusInterface, svcClient clientv1.ServiceInterface, selectorLabels map[string]string) error {
```

Comment:
it should be possible to write unit tests for this function.

Reply:
Could you point me to an example that uses mock clients?

Reply:
sure!

https://github.com/prometheus-operator/prometheus-operator/blob/b2a6a6ed8c33ee9a9a8eedd2809fbbe5895e2aba/pkg/prometheus/resource_selector_test.go#L1079-L1102

Reply:
Thanks!